### PR TITLE
Characterization & Pathways View Optimization

### PR DESCRIPTION
**Pull request by AntonStepanof**
fixes https://github.com/OHDSI/WebAPI/issues/828

### CODE REVIEW

**Comment by chrisknoll** [comment]
Why was the migration script called `V2.8.0.20190319114500__delete_design_column_from_views.sql` for oracle/postgresql, but `V2.8.0.20190319114500__sources-endpoint-permission.sql` on sqlserver?

**Comment by AntonStepanof** [comment]
> Why was the migration script called `V2.8.0.20190319114500__delete_design_column_from_views.sql` for oracle/postgresql, but `V2.8.0.20190319114500__sources-endpoint-permission.sql` on sqlserver?

Yes, It was wrong naming. Fixed

File: src/main/java/org/ohdsi/webapi/cohortcharacterization/CcServiceImpl.java [hunk 1]
```diff
@@ -354,7 +355,9 @@ public CohortCharacterizationEntity findByIdWithLinkedEntities(final Long id) {
     @Override
     @DataSourceAccess
     public CohortCharacterization findDesignByGenerationId(@CcGenerationId final Long id) {
-        return ccGenerationRepository.findById(id).map(gen -> gen.getDesign()).orElse(null);
+        final AnalysisGenerationInfoEntity generationInfoEntity = analysisGenerationInfoEntityRepository.findOne(id);
+        return generationInfoEntity != null ?
+                new SerializedCcToCcConverter().convertToEntityAttribute(generationInfoEntity.getDesign()) : null;
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/cohortcharacterization/CcServiceImpl.java]: You should not call `SerializedCcToCcConverter` here but use `GenericConvertionService`

File: src/main/java/org/ohdsi/webapi/cohortcharacterization/CcServiceImpl.java [hunk 2]
```diff
@@ -354,7 +355,9 @@ public CohortCharacterizationEntity findByIdWithLinkedEntities(final Long id) {
     @Override
     @DataSourceAccess
     public CohortCharacterization findDesignByGenerationId(@CcGenerationId final Long id) {
-        return ccGenerationRepository.findById(id).map(gen -> gen.getDesign()).orElse(null);
+        final AnalysisGenerationInfoEntity generationInfoEntity = analysisGenerationInfoEntityRepository.findOne(id);
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/cohortcharacterization/CcServiceImpl.java]: Why do you want to change `findById` to `findOne`?

File: src/main/java/org/ohdsi/webapi/pathway/PathwayServiceImpl.java [hunk 3]
```diff
@@ -400,7 +401,9 @@ public PathwayAnalysisResult getResultingPathways(final @PathwayAnalysisGenerati
 
         List<PathwayGenerationStats> targetCohortStatsList = queryGenerationStats(source, generationId);
 
-        generation.getDesign().getTargetCohorts().forEach(tc -> {
+        PathwayAnalysisEntity design = new SerializedPathwayAnalysisToPathwayAnalysisConverter()
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/pathway/PathwayServiceImpl.java]: Direct call to converter from service looks very nasty,
and why `PathwayAnalysisGenerationEntity` is not related with `AnalysisGenerationInfoEntity`?

File: src/main/java/org/ohdsi/webapi/pathway/PathwayServiceImpl.java [hunk 4]
```diff
@@ -446,6 +449,13 @@ public PathwayAnalysisResult getResultingPathways(final @PathwayAnalysisGenerati
         return result;
     }
 
+    @Override
+    @DataSourceAccess
+    public String findDesignByGenerationId(@PathwayAnalysisGenerationId final Long id) {
+        final AnalysisGenerationInfoEntity generationInfoEntity = analysisGenerationInfoEntityRepository.findOne(id);
```
Comment (wivern) [inline src/main/java/org/ohdsi/webapi/pathway/PathwayServiceImpl.java]: Why not to use `Optional.orElse` ?

File: src/main/java/org/ohdsi/webapi/cohortcharacterization/CcServiceImpl.java (hunk 1 above)
Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/cohortcharacterization/CcServiceImpl.java]: done

File: src/main/java/org/ohdsi/webapi/cohortcharacterization/CcServiceImpl.java (hunk 2 above)
Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/cohortcharacterization/CcServiceImpl.java]: done

File: src/main/java/org/ohdsi/webapi/pathway/PathwayServiceImpl.java (hunk 3 above)
Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/pathway/PathwayServiceImpl.java]: I changed that to calling GenericConversionService

File: src/main/java/org/ohdsi/webapi/pathway/PathwayServiceImpl.java (hunk 4 above)
Comment (AntonStepanof) [reply src/main/java/org/ohdsi/webapi/pathway/PathwayServiceImpl.java]: done

**Pull request merged**